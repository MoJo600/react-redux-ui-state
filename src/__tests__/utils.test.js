import identity from 'lodash/identity';
import { generateMaps, generateName, generateType, types } from '../';

jest.mock('uuid/v4', () => () => '110ec58a-a0f2-4ac4-8393-c866d813b8d1');

describe('ui state lib utils', () => {
    const name = generateName('action-test-component');
    const maps = generateMaps(name);

    it('should make a consistent name', () => {
        expect(generateName('test-component-name')).toMatchSnapshot();
        expect(generateName()).toMatchSnapshot();
    });

    it('should keep consistent types', () => {
        expect(types).toMatchSnapshot();
    });

    it('should create correct types', () => {
        expect(generateType(types.add, name)).toMatchSnapshot();
        expect(generateType(types.set, name)).toMatchSnapshot();
    });

    it('should create a maps object', () => {
        expect(maps.dispatchToProps).toMatchSnapshot();
        expect(maps.stateToProps).toMatchSnapshot();
    });

    it('should dispatch the add action', () => {
        const mockDispatch = jest.fn(identity);
        const dispatchToProps = maps.dispatchToProps(mockDispatch);
        const add = dispatchToProps.add({ val1: true, val2: false });

        expect(add).toMatchSnapshot();
        expect(mockDispatch.mock.calls.length).toBe(1);
    });

    it('should dispatch the set action', () => {
        const mockDispatch = jest.fn(identity);
        const dispatchToProps = maps.dispatchToProps(mockDispatch);
        const add = dispatchToProps.set({ val1: false, val2: true });

        expect(add).toMatchSnapshot();
        expect(mockDispatch.mock.calls.length).toBe(1);
    });
});
