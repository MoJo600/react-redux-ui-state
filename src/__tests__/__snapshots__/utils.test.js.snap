// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui state lib utils should create a maps object 1`] = `[Function]`;

exports[`ui state lib utils should create a maps object 2`] = `[Function]`;

exports[`ui state lib utils should create correct types 1`] = `"UI_STATE_ADD:action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1"`;

exports[`ui state lib utils should create correct types 2`] = `"UI_STATE_SET:action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1"`;

exports[`ui state lib utils should dispatch the add action 1`] = `
Object {
  "payload": Object {
    "name": "action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
    "state": Object {
      "val1": true,
      "val2": false,
    },
  },
  "type": "UI_STATE_ADD:action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
}
`;

exports[`ui state lib utils should dispatch the delete action 1`] = `
Object {
  "payload": Object {
    "name": "action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
  },
  "type": "UI_STATE_DELETE:action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
}
`;

exports[`ui state lib utils should dispatch the set action 1`] = `
Object {
  "payload": Object {
    "name": "action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
    "state": Object {
      "val1": false,
      "val2": true,
    },
  },
  "type": "UI_STATE_SET:action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1",
}
`;

exports[`ui state lib utils should keep consistent types 1`] = `
Object {
  "add": "UI_STATE_ADD",
  "delete": "UI_STATE_DELETE",
  "reset": "UI_STATE_RESET",
  "set": "UI_STATE_SET",
}
`;

exports[`ui state lib utils should make a consistent name 1`] = `"test-component-name_110ec58a-a0f2-4ac4-8393-c866d813b8d1"`;

exports[`ui state lib utils should make a consistent name 2`] = `"component_110ec58a-a0f2-4ac4-8393-c866d813b8d1"`;
